include/padscore/wpad.h

```diff
@@ -267,41 +457,345 @@ WPADInit();
+
+/**
+ * Writes to the registers of the Wii Remote's peripherals
+ * \param address address within the peripheral's memory space
+ * \sa
+ * - WPADWriteMemoryAsync()
+ * - WPADReadExtReg()
+ *
+ * Usage:
+ * \code
+ * // Setting speaker volume on specific controller
+ * uint8_t volume;
+ * volume = 0x40;
+ * WPADWriteExtReg(WPAD_CHAN_0, &volume, 1, WPAD_PERIPHERAL_SPACE_SPEAKER, 0x05, nullptr);
+ * \endcode
+ */
+int32_t
+WPADWriteExtReg(WPADChan channel,
+                void *source,
+                uint32_t size,
+                WPADPeripheralSpace peripheral,
+                uint32_t address,
+                WPADWriteMemoryCallback callback);
+
+/**
+ * Read Balance Board calibration
+ * \param address may only be in the ranges [0x24,0x40) and [0x50, 0x70)
+ * \sa
+ * - WPADReadExtReg()
+ */
+int32_t
+WPADGetBLCalibration(WPADChan channel,
+                     void *destination,
+                     uint32_t address,
+                     uint32_t size,
+                     WPADReadMemoryCallback callback);
+/**
+ * Sets power save mode, this makes the controller only report input data
+ * when it changes
+ */
 void
-WPADDisconnect(WPADChan chan);
+WPADSetPowerSaveMode(WPADChan chan,
+                     BOOL powerSave);
 
+/**
+ * \return FALSE, if power save mode is off
+ */
+BOOL
+WPADGetPowerSaveMode(WPADChan chan);
+
+/**
+ * Retrieves the bluetooth address of the controller
+ */
+void
+WPADGetAddress(WPADChan chan,
+               WPADAddress* outAddress);
+
+/**
+ * Enables/disables motors globally
+ */
+void 
+WPADEnableMotor(BOOL enable);
+
+/**
+ * Returns whether motors are disabled globally
+ */
+BOOL 
+WPADIsMotorEnabled();
+
+/**
+ * Enables/disables Wii U Pro Controller support
+ */
+void
+WPADEnableURCC(BOOL enable);
+
+/**
+ * Returns whether Wii U Pro Controllers are supported
+ */
+BOOL
+WPADIsEnabledURC();
+
+/**
+ * Enables/disables Wii Balance Board support
+ */
+void
+WPADEnableWBC(BOOL enable);
+
+/**
+ * Returns whether Wii Balance Boards are supported
+ */
+BOOL
+WPADIsEnableWBC();
+
+/**
+ * Enables/disables Wii Remote support
+ */
+void
+WPADEnableWiiRemote(BOOL enable);
+
+void
+WPADSetAutoSleepTime(uint8_t time);
+
+/**
+ * Starts searching for a WPAD controller in pairing mode and syncs with it
+ * \return TRUE if sync started
+ */
+BOOL
+WPADStartSyncDevice();
+
+/**
+ * Starts searching for a WPAD controller in pairing mode that matches specified properties, and syncs with it
+ * \param deviceAddress Bluetooth address of the device to connect to. If deviceAddress is 00:00:00:00:00:00, any address is matched
+ * \param deviceName up to 24 starting characters of the Bluetooth name of the device to connect to, full name not necessary
+ * \return TRUE if sync started
+ *
+ * Usage:
+ * \code
+ * WPADAddress addr;
+ * memset(&addr, 0, 6);
+ * // Will search for controllers with any address and a name that starts with "Nintendo"
+ * WPADStartSyncDevice(addr, "Nintendo");
```

Comment:
I don't think this usage example works.
On the IOSU-side the name needs to start with `Nintendo RVL-CNT-01-UC`, `Nintendo RVL-CNT` or `Nintendo RVL-WBC` and the BDA is used to retrieve the device info which would fail if it's zeroed.

Reply:
> The device is also not paired if the name is not at least a partial match

I'm able to pair my Wii Remote with the following code without any issues:
```c
    WPADAddress addr;
    memset(&addr, 0xab, 6);
    WPADStartSyncDeviceEx(&addr, "Potato");
```
I wonder if it will start a normal sync, if a sync with the provided parameters fails.

Also your example code is missing an `&` before `addr` here: `WPADStartSyncDeviceEx(addr, "Nintendo");`.